src/composer.rs

```diff
@@ -355,15 +397,16 @@ impl ContainerSpec {
         vec
     }
     /// Command/entrypoint followed by/and arguments
-    fn commands(&self, network: &str) -> Vec<String> {
+    fn commands(&self, network: &str) -> Vec<OsString> {
```

Comment:
I wonder if returning `String` would confine the unwraps here avoid the `to_str().unwrap()` elsewhere.

Reply:
That's true. That's actually the whole of having OsString: there are not the same as String. They are platform dependent, and it is possible that a string valid for a particular platform cannot be converted into a Rust String (which is a strictly valid UTF-8 sequence).

That is, the recommended practice is use OsString/OsStr for any platform-dependent strings: file names, shell command arguments, environment variables values and their names, etc, etc.
If it is not just a string but a file system path, PathBuf and Path should be used instead of OsString/OsStr. It hides details about concatenating paths (/ vs \), etc.